### PR TITLE
added detail json for addons

### LINKED ISSUE #15862: implement "detailed" addon list for -o=json
### What Happened?

minikube addon list has a flag called -d that will show extra information
```
$ minikube addons list  -d
|-----------------------------|----------|--------------|--------------------------------|-----------------------------------------------------------------------------------|
|         ADDON NAME          | PROFILE  |    STATUS    |           MAINTAINER           |                                       DOCS                                        |
|-----------------------------|----------|--------------|--------------------------------|-----------------------------------------------------------------------------------|
| ambassador                  | minikube | disabled     | 3rd party (Ambassador)         | https://minikube.sigs.k8s.io/docs/tutorials/ambassador_ingress_controller/        |
| auto-pause                  | minikube | disabled     | Google                         | n/a                                                                               |
| cloud-spanner               | minikube | disabled     | Google                         | https://minikube.sigs.k8s.io/docs/handbook/addons/cloud-spanner/                  |
| csi-hostpath-driver         | minikube | disabled     | Kubernetes                     | https://minikube.sigs.k8s.io/docs/tutorials/volume_snapshots_and_csi/             |
| dashboard                   | minikube | disabled     | Kubernetes                     | https://minikube.sigs.k8s.io/docs/handbook/dashboard/                             |
| default-storageclass        | minikube | enabled ✅   | Kubernetes                     | https://minikube.sigs.k8s.io/docs/handbook/persistent_volumes/                    |
| efk                         | minikube | disabled     | 3rd party (Elastic)            | n/a                                                                               |
| freshpod                    | minikube | disabled     | Google                         | https://github.com/GoogleCloudPlatform/freshpod                                   |
| gcp-auth                    | minikube | disabled     | Google                         | https://minikube.sigs.k8s.io/docs/handbook/addons/gcp-auth/                       |

```

but this information is not implemented in the json format of the command
```
$ minikube addons list -o=json -d
{"ambassador":{"Profile":"minikube","Status":"disabled"},"auto-pause":{"Profile":"minikube","Status":"disabled"},"cloud-spanner":{"Profile":"minikube","Status":"disabled"},"csi-hostpath-driver":{"Profile":"minikube","Status":"disabled"},"dashboard":{"Profile":"minikube","Status":"disabled"},"default-storageclass":{"Profile":"minikube","Status":"enabled"},"efk":{"Profile":"minikube","Status":"disabled"},"freshpod":{"Profile":"minikube","Status":"disabled"},"gcp-auth":{"Profile":"minikube","Status":"disabled"},"gvisor":{"Profile":"minikube","Status":"disabled"},"headlamp":{"Profile":"minikube","Status":"disabled"},"helm-tiller":{"Profile":"minikube","Status":"disabled"},"inaccel":{"Profile":"minikube","Status":"disabled"},"ingress":{"Profile":"minikube","Status":"disabled"},"ingress-dns":{"Profile":"minikube","Status":"disabled"},"istio":{"Profile":"minikube","Status":"disabled"},"istio-provisioner":{"Profile":"minikube","Status":"disabled"},"kong":{"Profile":"minikube","Status":"disabled"},"kubevirt":{"Profile":"minikube","Status":"disabled"},"logviewer":{"Profile":"minikube","Status":"disabled"},"metallb":{"Profile":"minikube","Status":"disabled"},"metrics-server":{"Profile":"minikube","Status":"disabled"},"nvidia-driver-installer":{"Profile":"minikube","Status":"disabled"},"nvidia-gpu-device-plugin":{"Profile":"minikube","Status":"disabled"},"olm":{"Profile":"minikube","Status":"disabled"},"pod-security-policy":{"Profile":"minikube","Status":"disabled"},"portainer":{"Profile":"minikube","Status":"disabled"},"registry":{"Profile":"minikube","Status":"disabled"},"registry-aliases":{"Profile":"minikube","Status":"disabled"},"registry-creds":{"Profile":"minikube","Status":"disabled"},"storage-provisioner":{"Profile":"minikube","Status":"enabled"},"storage-provisioner-gluster":{"Profile":"minikube","Status":"disabled"},"volumesnapshots":{"Profile":"minikube","Status":"disabled"}}
```

we need to add this missing fields for -d when users chooses -o=json

### Attach the log file

n/a

### Operating System

None

### Driver

None

### PR DESCRIPTION
**Pull request by swastik959**
**added detail json for addons**

- fixes #15862 
**before**
Screenshot from 2023-02-16 16-24-09

**After**
Screenshot from 2023-02-18 10-27-45

### CODE REVIEW

**Comment by medyagh** [comment]
Thank u for this contribution can u please put the output of minikube before/after this pr

**Comment by swastik959** [comment]
> Thank u for this contribution can u please put the output of minikube before/after this pr

@medyagh attached the screenshots . are these the changes you wanted

File: cmd/minikube/cmd/config/addons_list.go [hunk 1]
```diff
@@ -166,13 +166,19 @@ var printAddonsJSON = func(cc *config.ClusterConfig) {
...
 		addonBundle := assets.Addons[addonName]
 		enabled := addonBundle.IsEnabled(cc)
-
+	        maintainer := addonBundle.Maintainer
+		docs := addonBundle.Docs
 		addonsMap[addonName] = map[string]interface{}{
 			"Status":  stringFromStatus(enabled),
 			"Profile": cc.Name,
+			"maintainer": maintainer,
+			"docs":       docs,
```
Comment (spowelljr) [inline cmd/minikube/cmd/config/addons_list.go:175]:
```suggestion
			"Maintainer": maintainer,
			"Docs":       docs,
```
For consistency sake we should keep the keys starting with a capital.

File: cmd/minikube/cmd/config/addons_list.go [hunk 2]
```diff
@@ -166,13 +166,19 @@ var printAddonsJSON = func(cc *config.ClusterConfig) {
...
 		}
 	}
-	jsonString, _ := json.Marshal(addonsMap)
-
-	out.String(string(jsonString))
+	jsonOut, err := json.MarshalIndent(addonsMap, "", "    ")
+	if err != nil {
+		exit.Error(reason.InternalJSONMarshal, "Failed to marshal addons to JSON", err)
+	} 
+	fmt.Println(string(jsonOut))
```
Comment (spowelljr) [inline cmd/minikube/cmd/config/addons_list.go:182]:
```suggestion
	out.String(string(jsonOut))
```
This should be left as `out.String`

**Review by spowelljr — CHANGES_REQUESTED** [review]
This implementation of the PR doesn't respect the `-d` flag. The extra fields should only be shown if `-d` is passed, but currently it's always outputting the extra fields.

**Comment by swastik959** [comment]
@spowelljr although I have fixed it now but somehow first docs are printing

**Comment by spowelljr** [comment]
> @spowelljr although I have fixed it now but somehow first docs are printing

I'm not sure what you mean by this, could you explain

**Comment by swastik959** [comment]
@spowelljr see the attached image in the release notes in the after section

File: cmd/minikube/cmd/config/addons_list.go [hunk 3]
```diff
@@ -166,13 +166,30 @@ var printAddonsJSON = func(cc *config.ClusterConfig) {
...
+				
+			}
+			
+		}
+		if !addonPrintDocs{
+		        addonsMap[addonName] = map[string]interface{}{
+				"Status":  stringFromStatus(enabled),
+				"Profile": cc.Name,
+		
+		        }
```
Comment (spowelljr) [inline cmd/minikube/cmd/config/addons_list.go:186]:
```suggestion
		addonsMap[addonName] = map[string]interface{}{
                        "Status":  stringFromStatus(enabled),
                        "Profile": cc.Name,
                }
                if addonPrintDocs {
                        addonsMap[addonName]["Maintainer"] = addonBundle.Maintainer
                        addonsMap[addonName]["Docs"] = addonBundle.Docs
```
This code can be refactored to greatly reduce duplication and lines of code

**Review by spowelljr — CHANGES_REQUESTED** [review]

**Comment by spowelljr** [comment]
> @spowelljr see the attached image in the release notes in the after section

Maps in go are unordered so you can't guarantee the order. But it doesn't matter as this is JSON and only computers are going to use it and they also don't care about the order.

**Comment by spowelljr** [comment]
Also, please run `make test` before pushing changes up to ensure they pass linting and unit tests.

**Comment by swastik959** [comment]
@spowelljr how do I fix linting and unit-tests

**Comment by spowelljr** [comment]
To fix the linting run `gofmt -s -w .` the unit test is failing because you're using custom indenting with `json.MarshalIndent` so you either have to remove it or update the test.

**Comment by swastik959** [comment]
@spowelljr please review this pr now

**Review by medyagh — APPROVED** [review]

**Pull request merged**